datadog-logs-oci-orm/modules/logging/data.tf

```diff
@@ -0,0 +1,47 @@
+data "oci_logging_log_groups" "datadog_service_log_group" {
+    #Required
+    compartment_id = var.compartment_ocid
+
+    #Optional
+    display_name = "datadog-service-logs"
+}
+
+# Step 1: Fetch all compartments in the tenancy
+data "oci_identity_compartments" "all_compartments" {
```

Comment:
If this module runs for every logging compartment in a loop
in `datadog-logs-oci-orm/main.tf`, then is this step run every time. If yes, this can be minimized by running once